src/server/game/Chat/LanguageMgr.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _LANGUAGE_MGR_H
+#define _LANGUAGE_MGR_H
+
+#include "Define.h"
+#include "Language.h"
+#include <string>
+#include <unordered_map>
+
+class TC_GAME_API LanguageMgr
+{
+    typedef std::pair<uint32, uint8> WordKey;
+    typedef std::vector<char const*> WordList;
+    typedef std::unordered_map<WordKey, WordList> WordsMap;
```

Comment:
Try with an ordered_map and add a member to WordKey that is dbc Id, use that to sort keys, and see if that fixes bad translate

Reply:
What would be the usage of it then?

Currently I'm not using any order of WordsMap but finding exactly what I need, so sorting it won't matter at the moment.

Reply:
I tried to change the WordList to include the dbc id of the word and then sorted all the WordList after loading them.
Sadly it didn't change anything.

Reply:
That's how the old client did it. Unfortunately if that changed due to a DBC reorder that may mean they changed the hash function or bucket size.

Reply:
I remember it looks very similar between the 2 versions, so I went back and tried to see what's different; I then noticed it's exactly the same (for language id 1, word group of length 3).

Found a bug in the code and now it's working :) Thanks!